plugins/uncnd_branch_print.c

```diff
@@ -0,0 +1,71 @@
+/*
+  This file is part of MAMBO, a low-overhead dynamic binary modification tool:
+      https://github.com/beehive-lab/mambo
+
+  Copyright 2021 The University of Manchester
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+#ifdef PLUGINS_NEW
+
+#include <stdio.h>
+#include <assert.h>
+#include <locale.h>
+#include <inttypes.h>
+#include "../plugins.h"
+
```

Comment:
It does not have a global counter.

Reply:
Is a global counter needed?

Reply:
You need a global counter to keep the numbers of the branches of each thread so at the end you can print the total number of branches for the application.
Have a look at how the global counter is used in [`branch_count.c`](https://github.com/jkressel/mambo/blob/riscv-plugins/plugins/branch_count.c#L29) and how is handled in the [`post_inst_handler`](https://github.com/jkressel/mambo/blob/riscv-plugins/plugins/branch_count.c#L72).